d3a_api_client/rest_device.py

```diff
@@ -15,20 +16,20 @@
 
 class RestDeviceClient(APIClientInterface, RestCommunicationMixin):
 
-    def __init__(self, simulation_id, device_id,
-                 domain_name=domain_name_from_env,
-                 websockets_domain_name=websocket_domain_name_from_env,
+    def __init__(self, device_id=None, simulation_id=None,
+                 domain_name=None,
+                 websockets_domain_name=None,
                  autoregister=False, start_websocket=True,
                  sim_api_domain_name=None):
-        self.simulation_id = simulation_id
+        self.simulation_id, self.domain_name, self.websockets_domain_name = \
```

Comment:
What's the use-case for this? If I had to instantiate a new `RestDeviceClient` I would expect to fetch my variables and then use them in the constructor:
```python
simulation_id, domain_name, websockets_domain_name = get_variables_from_file(filename)
RestDeviceClient(
    simulation_id=simulation_id,
    domain_name=domain_name,
    websockets_domain_name=websockets_domain_name)
```

Reply:
It is for user's convenience to not explicitly need to pass it but to have an option to auto read for the respective simulation config file.

Reply:
Hmm ok. I personally think it makes the class more complicated. But if you prefer it, that's good for me :)